configuration-rendering-controller/README.md

```diff
@@ -0,0 +1,481 @@
+namespace.
+
+We know a complete list of possible Apps that can be configured, because the current system requires that for all
+possible deployed App there are default in the `shared-configs` repository. We would generate the CM and the Secret
+for each possible App for the given MC and the given `shared-configs` + CCR repo reference in the destination namespace.
+
+The name of the CMs and Secrets should be generated in a deterministic way. We already sort of do that. For example
+for the App `kyverno` the `kyverno-konfigure` CM and Secret is getting rendered by `konfigure`.
+
+The CRD should be open to filter the Apps to generate the configuration for. This makes it possible to render subsets
+of the whole configuration to different namespaces. This is especially useful for Flux HelmReleases that do not
+support cross namespace references for values from CMs and Secrets.
+
+It is worth noting here, that the decoupling also comes with a generic problem in Kubernetes where there are dependencies
+between resources. Since in this new system, we can expect that configs are pre-rendered, the workloads can freely
+mount them. However, especially if the 2 CRs are deployed together, it takes time for the CM and Secret to get
+deployed. For Flux HelmReleases it will simply initially result in a failure as the mounted CM and / or Secret does
+not exist yet - unless marked as optional -, and simply a following reconciliation will eventually deploy the release.
+
+For App CRs however, we do have a rule in `app-admission-controller` that rejects an App CR if it references a CM or
+Secret under `.spec.config` or `.spec.userConfig` that does not exist. This should be relaxed, made optional or even
+better removed all together. We could also turn it into a warning - supported by admission controllers in kubernetes,
+see [here](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#response).
+In an ever-changing Kubernetes environment, it is fair to expect the system to become eventually consistent
+and self-healing. This is the practice in other operators as well. The problem should be raised when the App CR
+does not get eventually healthy, for which we already have alerts in place. Also, this does not raise a new problem
+but one we already have and is native to the ecosystem we are working on, just think of Flux and
+its CRDs with which it operates but also eventually manages itself with.
+
+Additionally, with the configuration system decoupled, App Platform could also be open for extension of being able to
+detect expected configurations or to react when they change. For example by supporting emitting Kubernetes event
+that App Platform or any component could watch and reconcile the resource earlier.
+
+In my opinion, it is strongly discouraged to automagically mount generated configuration for multiple reasons:
```

Comment:
I so not specifically agree on this one. Customer quite like the auto mounting for apps that are part of releases because it's harder to maintain them with flux

Reply:
For one, we can do that tho cos only a portion of apps using a specific deployment methods gets the configs now but there are many other apps. Also there are for example Helm Releases where we cannot add this to the controller but we still want it to take advantage of the config system.

For another, it is crucial that the config system does not do it / has no knowledge of the platform. Later, we can extend App Platform to mount the configs in some ways. E.g. we could support publishing events from the config system and app-operator could watch them. Or other ways too. It is a separate problem.

Reply:
That's true :)